### ISSUE TITLE
Card: provide `button` or `a` tag within the `card` component

### ISSUE THREAD

**Issue by mfranzke** [db-ux-design-system/core-web#212]
We should either use a `button` or `a` HTML tag within the `card` component to enable keyboard and screenreader compatibility.

_Originally posted by @mfranzke in https://github.com/db-ui/mono/pull/211#discussion_r1073673263_

**Comment by nmerget** [comment]
We should use `<button>` for interactive cards and then there should be a `disabled` prop for UX and tech
